### PR TITLE
Ladybird+Userland: Remove use of unnecessary fd passing socket concept using SCM_RIGHTS

### LINKED ISSUE #22357: LibIPC: Transfer file descriptors in out of band data on single socket
Currently, we use two sockets to transfer file descriptors and data in LibIPC for each service and application that needs to pass fds around.

This is the "fd passing socket" concept used for each server in Ladybird.

This pattern is cumbersome and we should be able to pass the sockets in out of band data on the unix sockets.

This needs serenity Kernel changes to match the behavior other Unixes, and userspace changes to adapt the LibIPC socket receive methods to match the 'standard' behavior. Serenity's kernel just queues up sendfd/recvfd pairs on the socket until you decide to fetch them. However, Linux/BSDs require you to pull the fd out of the socket at the same time you're receiving data or the socket is dropped on the floor.

### PR DESCRIPTION
**Pull request by ADKaster**
This refactor eliminates the need for a second "fd passing socket" on
Lagom, as it uses SCM_RIGHTS in the expected fashion, to send fds along
with the data of our Unix socket message.

Fixes #22357

### CODE REVIEW

**Comment by bugaevc** [comment]
Do you mean something specific that won't work on the Hurd? I don't immediately see anything that should break, although it would of course make sense for me to test.

**Comment by ADKaster** [comment]
Oh, wait, I misread the #ifdefs in the old LocalSocket::send_fd method. 

`#elif defined(AK_OS_LINUX) || defined(AK_OS_GNU_HURD) || defined(AK_OS_BSD_GENERIC) || defined(AK_OS_HAIKU)`

Serenity + These guys leaves.. what literally just Windows that this won't work on? which is meh to me, windows is for several years from now :sweat_smile:

File: Userland/Libraries/LibCore/Socket.h [hunk 1]
```diff
@@ -329,6 +329,11 @@ class LocalSocket final : public Socket {
 
     ErrorOr<int> receive_fd(int flags);
     ErrorOr<void> send_fd(int fd);
+
+    static constexpr size_t MAX_TRANSFER_FDS = 64;
```
Comment (trflynn89) [inline Userland/Libraries/LibCore/Socket.h:333]: nit: if this isn't needed outside of the below methods, I'd just define this locally in Socket.cpp. Much less of a rebuild if we change the value later on.

File: Userland/Libraries/LibWeb/HTML/MessagePort.h [hunk 2]
```diff
@@ -78,6 +78,12 @@ class MessagePort final : public DOM::EventTarget
     ErrorOr<void> send_message_on_socket(SerializedTransferRecord const&);
     void read_from_socket();
 
+    enum class ParseDecision {
+        NotEnoughData,
+        ParseNextMessage,
+    };
+    ErrorOr<ParseDecision> try_parse_message();
```
Comment (trflynn89) [inline Userland/Libraries/LibWeb/HTML/MessagePort.h:85]: I don't think we need the `try_` prefix when the fallible option is the only option

**Review by trflynn89 — COMMENTED** [review]
:star:

File: Userland/Libraries/LibCore/Socket.h (hunk 1 above)
Comment (ADKaster) [reply Userland/Libraries/LibCore/Socket.h]: moved to cpp

File: Userland/Libraries/LibWeb/HTML/MessagePort.h (hunk 2 above)
Comment (ADKaster) [reply Userland/Libraries/LibWeb/HTML/MessagePort.h]: tru, removed

**Review by trflynn89 — APPROVED** [review]

**Comment by awesomekling** [comment]
This is pure :yakkiss:

**Pull request merged**